crates/ruff_python_semantic/src/analyze/visibility.rs

```diff
@@ -42,21 +78,20 @@ pub fn is_override(decorator_list: &[Decorator], semantic: &SemanticModel) -> bo
 
 /// Returns `true` if a function definition is an abstract method based on its decorators.
 pub fn is_abstract(decorator_list: &[Decorator], semantic: &SemanticModel) -> bool {
-    decorator_list.iter().any(|decorator| {
-        semantic
-            .resolve_qualified_name(&decorator.expression)
-            .is_some_and(|qualified_name| {
-                matches!(
-                    qualified_name.segments(),
-                    [
-                        "abc",
-                        "abstractmethod"
-                            | "abstractclassmethod"
-                            | "abstractstaticmethod"
-                            | "abstractproperty"
-                    ]
-                )
-            })
+    abstract_decorator(decorator_list, semantic).is_some()
+}
+
+pub fn abstract_decorator<'a>(
+    decorator_list: &'a [Decorator],
+    semantic: &SemanticModel,
+) -> Option<(&'a Decorator, AbstractDecoratorKind)> {
```

Comment:
What's the reason for returning the `Decorator` here as I see that it's not being used? We can maybe just use `Option<AbstractDecoratorKind>` and add documentation like "Returns the abstract decorator kind for the function definition if it's an abstract method...".